core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp

```diff
@@ -2381,7 +2381,8 @@ SQLRETURN CDesc::BuildValueList(CStmt *pHandle,  //AMR - changed this pointer ty
 
 				if(IPDDescRecPtr->m_ODBCDataType == SQL_VARCHAR ||
 				   IPDDescRecPtr->m_ODBCDataType == SQL_LONGVARCHAR ||
-				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR)
+				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR &&
+                   IPDDescRecPtr->m_SQLDataType != SQLTYPECODE_BOOLEAN)
```

Comment:
when SQLDataType return SQLTYPECODE_BOOLEAN, ODBCDataType perform SQL_CHAR then code will make len-1.
It's unneccessary operation, so skip these code when meeting boolean type.
So VARCHAR or LONG VARCHAR or ( CHAR and not BOOLEAN)  is my meaning, and add () will be better